feature/DBI-US003-22 - US003 - [Front] - Validar dados

roazambuja opened the pull request:
## Descrição

Foi desenvolvida a validação de dados do formulário de cadastro de novo guia. Adicionamos os elementos necessários para validação de dados e realizamos os testes.


____

**Link do card:**
[US003 - [Front] - Validar dados](https://trello.com/c/vE5HlCDv/22-us003-front-validar-dados)

____

## Checklist Code Review
Avaliar se todos os itens a seguir foram feitos. Os itens com `*` não são obrigatórios.

- [x] Foi adicionada a descrição da PR
- [x] Foi adicionado o link do card
- [x] A branch segue o padrão com o prefixo DBI
- [x] Está seguindo os padrões de prefixo do gitflow (feature, bugfix, hotfix, release)
- [x] O título da PR segue o padrão: ***Nome da branch - Título do card***
- [x] A PR está sendo enviada para a branch `develop`
- [x] O Repositório que a PR está sendo enviada é o [**dbinclui-org/dbinclui-frontend**](https://github.com/dbinclui-org/dbinclui-frontend)
- [x] Foi realizado os testes unitários *
- [ ] Foi realizado os testes de E2E *
- [x] Foi adicionada as respectivas **labels**
- [x] A PR foi assinada

Para que haja uma padronização na criação dos componentes, este deve seguir o seguinte modelo de construção:

- [x] Deve ser feita a importação do _React_ no escopo do componente.
- [x] Deve conter uma _interface_ com as propriedades do componente.
- [x] Nome da _interface_ deve ter o sufixo _Props_.
- [x] Recebe _React.FC_, no qual recebe a _interface_
- [x] Deve retornar elemento _JSX_
- [x] O componente de ser exportado ao final como `default`.
 
Exemplo:
```tsx
import React from 'react';

export interface ComponetNameProps {}

export const ComponetName: React.FC<ComponentNameProps> = (): JSX.Element => {
  return <>...</>;
};

export default ComponentName;
```

roazambuja committed:
Alterando criação do componente

roazambuja committed:
Criação do formulário

roazambuja committed:
Validar dados do formulário

roazambuja committed:
Abstração de validador e confecção de testes

roazambuja committed:
Alterações na validação e nos testes

roazambuja committed:
Merge branch 'develop' of https://github.com/dbinclui-org/dbinclui-frontend into feature/DBI-US003-22

arthurvergacas committed:
[chore] Remove console.log and fix formatting

arthurvergacas committed:
[feature] Add return type to validateInput function

fabiojcbatista reviewed (commented):
Nome da branch--------------> feature/DBI-US003-22
Sufixo Props nas interface ---> export interface ComponetNameProps {}

eduardoaraya committed:
Create global scale to typography

arthurvergacas committed:
[fix] Add empty interface to follow requirements

arthurvergacas committed:
Merge branch 'develop' of https://github.com/dbinclui-org/dbinclui-frontend into feature/DBI-US003-22

arthurvergacas committed:
[fix] FIx build error

- MenuAcessibility was declared but never used in layout/index.tsx

roazambuja committed:
Remover console.log

roazambuja committed:
Alterar nome do arquivo de teste

roazambuja committed:
Alterações nos testes da função de validação

roazambuja committed:
Teste de exibição do formulário

arthurvergacas committed:
Adicionar biblioteca de testes user-event

Com ela é possível simular eventos que seriam gerados pelo usuário com maior precisão

arthurvergacas committed:
Instalar bibliotecas adicionais para user-event

arthurvergacas committed:
Testes de submit e de input a tela de registro de guia
